bin/BuildPackages.sh

```diff
@@ -52,86 +36,203 @@ build_packages() {
 # Even if it doesn't work completely automatically for you, you may get
 # an idea what to do for a complete installation of GAP.
 
+set -e
 
 # Is someone trying to run us from inside the 'bin' directory?
+run_from_bin(){
 if [[ -f gapicon.bmp ]]
 then
   error "This script must be run from inside the pkg directory" \
         "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
+}
+
+# set default parameters
+set_default_parameters() {
 
 # CURDIR
 CURDIR="$(pwd)"
 
-# GAPDIR
-GAPDIR=""
-if [[ "$#" -ge 1 ]]
+# default GAPDIR
+pushd .. >/dev/null
+GAPDIR="$(pwd)"
+popd >/dev/null
```

Comment:
Right, and it was an improvement over what you had before -- but I made the mistake to only think about the code that was already there, not to question what it was for, and then suggesting an alternative based on that :-)/